Framework/Algorithms/src/MaxEnt.cpp

```diff
@@ -677,14 +677,18 @@ void MaxEnt::populateImageWS(const MatrixWorkspace_sptr &inWS, size_t spec,
 
   int npoints = complex ? static_cast<int>(result.size() / 2)
                         : static_cast<int>(result.size());
-  int npointsX = inWS->isHistogramData() ? npoints + 1 : npoints;
-  MantidVec X(npointsX);
+  MantidVec X(npoints);
   MantidVec YR(npoints);
   MantidVec YI(npoints);
   MantidVec E(npoints, 0.);
 
-  double x0 = inWS->x(spec)[0];
-  double dx = inWS->x(spec)[1] - x0;
+  MantidVec dataPoints(npoints);
+  if (inWS->isHistogramData())
+    Kernel::VectorHelper::convertToBinCentre(inWS->readX(spec), dataPoints);
+  else
+    dataPoints = inWS->readX(spec);
```

Comment:
Since the introduction of the `HistogramData` module this can be done in a one-liner:

```cpp
auto dataPoints = inWS->points(spec);
```